native/src/main/java/io/ballerina/stdlib/file/nativeimpl/Utils.java

```diff
@@ -226,16 +234,29 @@ private static Object readFileTree(File inputFile) {
                     throw ErrorCreator.createError(StringUtils.fromString("Error while accessing file info"), e);
                 }
             }).skip(1).toArray(Object[]::new);
-            return ValueCreator.createArrayValue(results,
-                    TypeCreator.createArrayType(TypeCreator.createRecordType(FileConstants.METADATA,
-                            ModuleUtils.getModule(), 0, true, 0)));
+            return ValueCreator.createArrayValue(results, METADATA_ARRAY_TYPE);
         } catch (IOException | BError ex) {
             return FileUtils.getBallerinaError(FileConstants.FILE_SYSTEM_ERROR, ex);
         } catch (SecurityException ex) {
             return FileUtils.getBallerinaError(FileConstants.PERMISSION_ERROR, ex);
         }
     }
 
+    private static RecordType createMetaDataType() {
+        Type utc = TypeCreator.createTupleType(List.of(PredefinedTypes.TYPE_INT, PredefinedTypes.TYPE_DECIMAL),
+                PredefinedTypes.TYPE_NEVER, 0, true);
+        Map<String, Field> fields = Map.of(
+                "absPath", TypeCreator.createField(PredefinedTypes.TYPE_STRING, "absPath", 0),
+                "size", TypeCreator.createField(PredefinedTypes.TYPE_INT, "size", 0),
+                "modifiedTime", TypeCreator.createField(utc, "modifiedTime", 0),
+                "dir", TypeCreator.createField(PredefinedTypes.TYPE_BOOLEAN, "dir", 0),
+                "readable", TypeCreator.createField(PredefinedTypes.TYPE_BOOLEAN, "readable", 0),
+                "writable", TypeCreator.createField(PredefinedTypes.TYPE_BOOLEAN, "writable", 0)
+        );
```

Comment:
So we need to change this whenever we change the `Metadata` record type? Can we make them constants and add a comment, please?

Reply:
Unfortunately yes. Fixed